### PR TITLE
feat: Add signature to asset rented event

### PR DESCRIPTION
**Pull request by fzavalia**
Created a RentParams struct as a workaround for the stack too deep error

### CODE REVIEW

File: contracts/Rentals.sol [hunk 1]
```diff
@@ -379,34 +393,37 @@ contract Rentals is NonceVerifiable, NativeMetaTransaction, IERC721Receiver {
...
-        address _tenant,
-        address _contractAddress,
-        uint256 _tokenId,
-        bytes32 _fingerprint,
-        uint256 _pricePerDay,
-        uint256 _rentalDays,
-        address _operator
-    ) private {
-        IERC721Rentable asset = IERC721Rentable(_contractAddress);
+    function _rent(RentParams memory params) private {
```
Comment (nachomazzara) [inline contracts/Rentals.sol:411]: can we call the variable `rentParam`. `params` is too generic

File: contracts/Rentals.sol [hunk 2]
```diff
@@ -97,6 +109,7 @@ contract Rentals is NonceVerifiable, NativeMetaTransaction, IERC721Receiver {
         address _operator,
         uint256 _rentalDays,
         uint256 _pricePerDay,
+        bytes _signature,
```
Comment (nachomazzara) [inline contracts/Rentals.sol:112]: I would move `_signature` to the end. It is going to be cheaper

Comment (fzavalia) [reply contracts/Rentals.sol:112]: 349944 → 349931 True :)

**Comment by fzavalia** [comment]
347207 → 349931 
2.7k more gas is required (for acceptListing) by sending the signature in the event and using the RentParams struct to bypass the stack to deep issue.

**Review by nachomazzara — APPROVED** [review]

**Pull request merged**